File: UI/src/app/shared/models/city.ts
import { Country } from "./country";
import { State } from "./state";

export interface City {
    id: string;
    name: string;
    countryId: string; // Assuming string type for foreign key, change to appropriate type if needed
    state: string; // Assuming string type for foreign key, change to appropriate type if needed
    stateId?: State | null;
    country?: Country | null;
    action:number;
  }
  
export interface SaveCity {
  id: string;
  name: string;
  countryId: string; // Assuming string type for foreign key, change to appropriate type if needed
  state: string; // Assuming string type for foreign key, change to appropriate type if needed
  stateId?: State | null;
  country?: Country | null;
  action:number;
}

File: API/beONHR.Entities/City.cs
using Microsoft.EntityFrameworkCore;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace beONHR.Entities
{
    
    public class City
    {
        [Key]
        public Guid Id { get; set; }

        public string Name { get; set; }
        public Guid CountryId { get; set; } //froginkey
        public Guid State { get; set; } //froginkey
        public bool IsDeleted { get; set; } = false;
        [ForeignKey("State")]
        public virtual State? stateId { get; set; }

        [ForeignKey("CountryId")]
        public virtual Country? country { get; set; }
    }
}
